### ISSUE TITLE
process is not defined

### ISSUE THREAD

**Issue by alinz** [rackt/redux#525]
Hello,
First of all I would like to give you a big thank.

So, I had no problem with 1.0.0-rc until I removed the node_modules and install everything again. what I found was red death screen which complains about `process is not defined`. At the first, I thought I missed something and I started debugging the code. 

Here's the code which causes the issue.

```
const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
const reducer = combineReducers(reducers);
const store = createStoreWithMiddleware(reducer);
```

`combineReducers` method cause the whole thing to crash. I have not changed any code on my side.

So has anyone have this problem? Have you guys pushed code into 1.0.0-rc branch? It used to work.

**Comment by chentsulin** [comment]
I believe this is because react-native's JavaScriptCore env doesn't have v8's `process` object. So we can use `__DEV__` instead.

And after some moment if this project will only need to support react-native version which include https://github.com/facebook/react-native/pull/2279 polyfill. Then, we can drop all of `__DEV__`.

**Comment by gaearon** [comment]
Argh, this is really frustrating. Second time I'm burned by this.

**Comment by gaearon** [comment]
Should be fixed in 1.0.1.

**Comment by alinz** [comment]
Awesome, Thank you @gaearon!
